ui/src/app/utils/index.tsx

```diff
@@ -277,6 +277,16 @@ export const withCurrentWorkspace = () => {
   return connectBehaviorSubject(currentWorkspaceStore, 'workspace');
 };
 
+// HOC that provides a 'workspace' prop with current WorkspaceData
+export const withCurrentCohort = () => {
+  return connectBehaviorSubject(currentCohortStore, 'cohort');
+};
+
+// HOC that provides a 'workspace' prop with current WorkspaceData
+export const withCurrentConceptSet = () => {
```

Comment:
Tangent: General question on these HOC stores. (In cases like the breadcrumb this question is probably not applicable since it stays on the page, but more a question for the other content below the nav bar.)

Does this always work cleanly when there's a transition between contexts? For example, would this ever be rendered before the currentWorkspace() updated on a navigate - or similarly would currentWorkspace() ever trigger a state change during navigation while this component was still on the screen?

Reply:
It depends on the exact case, but generally yes. React (and I think Angular too) unmounts components from leaves upward, so as long as the state in question is being managed by a component higher up in the tree, the child will be removed before it sees any changes.

Obviously if we're making a lateral move, e.g. going from one participant to the next, then the component will stay mounted and see the transition, but the component expects that.